test/ThriftTest.thrift

```diff
@@ -411,3 +411,17 @@ struct StructB {
 struct OptionalSetDefaultTest {
   1: optional set<string> with_default = [ "test" ]
 }
+
+service ConflictingNamesTest {
```

Comment:
The ThriftTest.thrift file serves a very specific purpose: It is the base for the cross language Test Suite. 
Adding unrelated tests in that file is highly discouraged.

Reply:
OK I'll create another PR to move this part out to a different thrift file, and if needed we can cherry pick that to 0.14.0 branch as well.

Reply:
https://github.com/apache/thrift/pull/2334

Reply:
Also cherry picked into this branch (thus added into this PR).